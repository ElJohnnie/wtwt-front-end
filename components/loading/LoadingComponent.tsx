import React from 'react';
import Lottie from 'lottie-react';
import loadingLottie from '../../assets/loading-lottie.json';

const LoadingComponent = ({ width = 450, height = 450 }) => {
  return (
    <div style={{ width: width, height: height }}>
      <Lottie animationData={loadingLottie} loop={true} />
    </div>
  );
};

export default LoadingComponent;
