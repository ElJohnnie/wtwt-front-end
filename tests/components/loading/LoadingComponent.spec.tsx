import React from 'react';
import { render } from '@testing-library/react';
import LoadingComponent from '../../../components/loading/LoadingComponent';

jest.mock('lottie-react', () => ({
  __esModule: true,
  default: () => <div>Lottie Mock</div>,
}));

describe('LoadingComponent', () => {
  test('renders correctly', () => {
    const { container } = render(<LoadingComponent />);

    expect(container).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
