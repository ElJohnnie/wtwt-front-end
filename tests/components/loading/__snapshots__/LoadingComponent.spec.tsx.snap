// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingComponent renders correctly 1`] = `
<div>
  <div
    style="width: 450px; height: 450px;"
  >
    <div>
      Lottie Mock
    </div>
  </div>
</div>
`;
